python/cudf/cudf/core/column/numerical.py

```diff
@@ -282,48 +282,42 @@ def nans_to_nulls(self: Self) -> Self:
+    def _normalize_binop_operand(self, other: Any) -> pa.Scalar | ColumnBase:
         if isinstance(other, ColumnBase):
             if not isinstance(other, type(self)):
                 return NotImplemented
             return other
-        if isinstance(other, cudf.Scalar):
-            if self.dtype == other.dtype:
-                return other
-
-            # expensive device-host transfer just to
-            # adjust the dtype
-            other = other.value
-
-            # NumPy 2 needs a Python scalar to do weak promotion, but
-            # pandas forces weak promotion always
-            # TODO: We could use 0, 0.0, and 0j for promotion to avoid copies.
-            if other.dtype.kind in "ifc":
-                other = other.item()
-        elif not isinstance(other, (int, float, complex)):
-            # Go via NumPy to get the value
-            other = np.array(other)
-            if other.dtype.kind in "ifc":
-                other = other.item()
-
-        # Try and match pandas and hence numpy. Deduce the common
-        # dtype via the _value_ of other, and the dtype of self on NumPy 1.x
-        # with NumPy 2, we force weak promotion even for our/NumPy scalars
-        # to match pandas 2.2.
-        # Weak promotion is not at all simple:
-        # np.result_type(0, np.uint8)
-        #   => np.uint8
-        # np.result_type(np.asarray([0], dtype=np.int64), np.uint8)
-        #   => np.int64
-        # np.promote_types(np.int64(0), np.uint8)
-        #   => np.int64
-        # np.promote_types(np.asarray([0], dtype=np.int64).dtype, np.uint8)
-        #   => np.int64
-        common_dtype = np.result_type(self.dtype, other)  # noqa: TID251
-        if common_dtype.kind in {"b", "i", "u", "f"}:
-            if self.dtype.kind == "b":
-                common_dtype = min_signed_type(other)
-            return cudf.Scalar(other, dtype=common_dtype)
+        elif is_scalar(other):
+            if is_na_like(other):
+                return super()._normalize_binop_operand(other)
+            if not isinstance(other, (int, float, complex)):
+                # Go via NumPy to get the value
+                other = np.array(other)
+                if other.dtype.kind in "uifc":
+                    other = other.item()
+
+            # Try and match pandas and hence numpy. Deduce the common
+            # dtype via the _value_ of other, and the dtype of self on NumPy 1.x
+            # with NumPy 2, we force weak promotion even for our/NumPy scalars
+            # to match pandas 2.2.
+            # Weak promotion is not at all simple:
+            # np.result_type(0, np.uint8)
+            #   => np.uint8
+            # np.result_type(np.asarray([0], dtype=np.int64), np.uint8)
+            #   => np.int64
+            # np.promote_types(np.int64(0), np.uint8)
+            #   => np.int64
+            # np.promote_types(np.asarray([0], dtype=np.int64).dtype, np.uint8)
+            #   => np.int64
```

Comment:
Doesn't have to be done in this PR, but we should check if there are better ways to work with this now...